src/definition/rooms/IPreRoomUserJoined.ts

```diff
@@ -0,0 +1,16 @@
+import { IHttp, IPersistence, IRead } from '../accessors';
+import { AppsEngineError } from '../errors';
+import { IUser } from '../users';
+import { IRoom } from './IRoom';
+
+export interface IPreRoomUserJoined {
+    executePreRoomUserJoined(context: IPreRoomUserJoinedContext, read: IRead, http: IHttp, persistence: IPersistence): Promise<void>;
+}
+
+export interface IPreRoomUserJoinedContext {
+    joiningUser: IUser;
+    room: IRoom;
+    invitingUser?: IUser;
```

Comment:
What if we name this as `actor`? This could be used later to identity who executed for other actions. Could even have more info and the user be one of the optional properties, could identify if it was a Cron action for example.

Reply:
Not so sure about it... It sounds too abstract. Even though it _could_ be beneficial in some way regarding reusability (which is still unclear for me), the concepts of both actions/events you mentioned are _very_ different. If we don't have a clear understanding of what that `actor` represents and what information it shares with other contexts, it might turn out to be harmful down the road

Reply:
I don't like the name `invitingUser` at all, it confusing to me, IMO it should be more generic indeed

Reply:
I'm not sure why it causes confusion.

Maybe simply rewording it should be enough?

Reply:
Please try, I'm open for suggestions and waiting. I did my already :)
I'm confused because I'm not quite sure if `invitingUser` is who invited or who was invited.

Reply:
Changed it to `inviter`